### DISCUSSION TITLE
unable to edit / view tables with BB made pages

### DISCUSSION THREAD

**Discussion by KernOil** [Budibase/budibase#2069] [Help]
Hi  @GuilhermeHirsch, I am having a problem when I use the auto-generated pages viewing a view from MySQL database the table populated is correct however if you select the view button it will only take me to edit the same row no matter which row I select. and none of the edits will save. 

example:
I generate a new screen with the built in options 
![image](https://user-images.githubusercontent.com/80058238/126718387-1f702e04-b793-47ef-8b28-f1fac254fa16.png)
![image](https://user-images.githubusercontent.com/80058238/126718390-51e93220-3953-4aa0-9d25-66ed899d0443.png)

I then get this page and table, when selecting any of the view buttons I go to the exact same page. 

![image](https://user-images.githubusercontent.com/80058238/126718456-06669ab7-0ee6-47d3-b3cc-178b0be998bd.png)

even throw the data is different on each line I only get this one line to edit and it will not save 

![image](https://user-images.githubusercontent.com/80058238/126718536-86cbbe4b-79af-4585-921d-9980c9a10b15.png)

please let me know if i am doing something wrong or if this is a bug that is being fixed

**Comment by mike12345567** [comment]
Hi @AndrewJRamay is there any errors when you save? For example if you open up the developer console and have a look at the network tab, see if there is any errors occurring when it attempts to save the row.

Also, out of interest does your table have a primary key set? Budibase requires all SQL databases to have primary key constraints specified for each table.

**Reply by KernOil** [reply]
no, it does not have a primary key. the only reason for this is because BB can not read the tables with the primary key it will give me an error and stops writing to the table.

**Reply by KernOil** [reply]
this is what i always get with a key row 
![image](https://user-images.githubusercontent.com/80058238/126721498-44cd925b-adea-42df-a6d5-ac8fb0cacaed.png)

**Reply by mike12345567** [reply]
Hi @AndrewJRamay - i'm a little confused I have a couple of questions about your database setup:
1. The error you showed, where do you get this error? What do you need to do for it to occur?
2. When you say it cannot read the tables with the primary key what do you mean? For Budibase to assess your SQL tables and work from them in its opinionated way every table has to have a primary key.
3. If possible, it might clear up all my questions, could you send through a picture of your mySQL configuration page? With the list of tables on it?

**Reply by KernOil** [reply]
@mike12345567 
1) for the error to show up you will need to have a table set up in MySQL with a primary key. to get the error go to the data table and try and add a new row to that new table. the primary key will be a data field. if you can accurately guess and input the correct primary key the row will save if you have the wrong key input it will fail. to my knowledge, I don't think we should see the primary key in the table at all as an input field. I believe this is where the error occurs. 

2) I'm sorry I meant to say you can no longer submit data to that table. yes BB will read the table but it will be a static table. 

3) here are the two tables I have.
![image](https://user-images.githubusercontent.com/80058238/126840142-afc763b6-e3d9-445e-ab6f-f742e3f5085e.png)

here is my test table i have set up with id as the primary key. 

![image](https://user-images.githubusercontent.com/80058238/126840218-de9c130c-3eeb-400e-89b3-c4be7cf081d6.png)

when I create a row I see the primary key as an input. 


![image](https://user-images.githubusercontent.com/80058238/126840303-fc5574f8-ea86-442b-b1fb-f5591f4f7cd3.png)

here BB has identified the column as the primary key 

![image](https://user-images.githubusercontent.com/80058238/126840384-3718ff11-b791-43d1-9b4b-51c0145749ba.png)

if I try and submit a row I get this 

![image](https://user-images.githubusercontent.com/80058238/126840500-e713745b-9e58-40ae-b897-b2726431d524.png)

if I guess wrong at the key-value ill get this 

![image](https://user-images.githubusercontent.com/80058238/126840548-bd354327-d82d-4f91-ba8e-8d1b140da499.png)

**Reply by KernOil** [reply]
also, the date still won't save. that's a different error/problem entirely tho.
![image](https://user-images.githubusercontent.com/80058238/126840863-da075e4a-bab0-4986-98b3-c40f1b251329.png)
